vc/pokecrystal11_vc.patch.template

```diff
@@ -0,0 +1,698 @@
+Mode = 2
+Address = {HEX @ 4}
+Type = 12
+[to_play2_battle]
+Mode = 2
+Address = {HEX @ 4}
+Type = 13
+[ret_heya]
+Mode = 2
+Address = {HEX @ 4}
+Type = 14
+
+
+
+
+
+;ROM:3FBCD                 ld      b, $3E ; '>'
+;ROM:3FBCF                 inc     de
+;ROM:3FBD0                 call    unk_2D55
+;ROM:3FBD3                 xor     a
+;ROM:3FBD4                 ld      [byte_FFD2], a
+;ROM:3FBD6                 ld      [byte_FF4A], a
+;ROM:3FBD8                 call    unk_31C2
+;ROM:3FBDB                 call    unk_2FE2
+;
+;ROM:3FBCD:  06 3E 13 CD
+
+;0003fbb5h: 06 3E 13 CD                                     ; .>.?
+
+[fight begin]                        
+Mode = 11                       
+Type = 0                        
+Index = 1                       
+Address = {HEx @}            
+Fixcode={db SCREEN_HEIGHT_PX}        
+              
+;12 1b 0b 79 b0  find next C9
+[BiographySave_ret]
+Mode = 2
+Address = {HEX @ 4}
+Type = 60
+
+
+; print forbid 1
+;ROM:1758D                 ld      a, [byte_FFA9]
+;ROM:1758F                 and     2
+;ROM:17591                 jr      nz, unk_75B4
+;ROM:17593                 ld      a, [byte_FFA9]
+;ROM:17595                 and     1                      ;e6 01
+;ROM:17597                 jr      nz, unk_75A1
+;    
+; change "and 1" to "and 0"
+;00017595h: E6 01 20 08 CD BF 75 CD 2E 03 18 E9 FA 57 CE F5 ; ? .涂u?..辁W熙
+;00016c76h: E6 01 20 08 CD A0 6C CD 5A 04 18 E9 FA 63 CF F5 ; ? .蜖l蚙..辁c硝
+[print forbid 1]                                                   
+Mode = 1                                                                                                                  
+Address = {hex @ 4}
+Fixcode={patch}      
+
+
+                            
+[print forbid 2]                                                                           
+Mode = 6                                                                                       
+Type = 0                                                                                       
+Address = {hex @ 4}                                                                        
+MemAddress={hex hJoyPressed}                                                                           
+Fixcode={db NO_INPUT}                                                                                  
+ConditionType = 0                                                                              
+ConditionValueA = {dws_ wWindowStackPointer wWindowStackPointer+1 wMenuJoypad wMenuSelection wMenuSelection wMenuCursorY hJoyPressed hJoyPressed hJoyPressed hJoyPressed}                   
+ConditionValueB = {dws_ ==                  ==                    ==          >=             <=             ==           !=          !=          !=          !=         }                   
+ConditionValueC = {dws_ 0xdd                0xd3                  0x01        0x00           0x0f           0x03         D_DOWN      D_UP        B_BUTTON    NO_INPUT   }                   
```

Comment:
`wMenuJoypad == NO_INPUT`

Reply:
I think there was a reason I didn't set these to the buttons.. but I don't remember why. It might have been because I had a question about it, and forgot to ask. 

So, what exactly is the purpose of wMenuJoypad in game? If I remember correctly, while debugging it usually stays set to 0x00, and only very very briefly changes to something else when pressed in a menu. To me this seems a bit redundant. Unless it is trying to prevent issues of people "holding down buttons" too long; or something to that effect.

Reply:
The values it briefly changes to are button bitmasks, so IMO it should use the semantically correct constants in the template. Consider the (unlikely) scenario where someone swaps the nybble order in which the 8 buttons are stored in one byte.